src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -1100,7 +1100,9 @@ def visit_Subscript(self, node: ast.Subscript):
                 ]
                 if len(result.data_index) != len(self.fields[result.name].data_dims):
                     raise GTScriptSyntaxError(
-                        "Incorrect data index length", loc=gt_ir.Location.from_ast_node(node)
+                        f"Incorrect data index length {len(result.data_index)}. "
+                        f"Field {result.name} has {len(self.fields[result.name].data_dims)} data dimensions.",
```

Comment:
The message should indicate an error.
```suggestion
                        f"Invalid data dimension index. Field {result.name} has {len(self.fields[result.name].data_dims)} data dimensions.",
```